codegen-service/llm-generation-core/src/main/java/org/rj/modelgen/llm/prompt/TemplatedPromptGenerator.java

```diff
@@ -0,0 +1,54 @@
+package org.rj.modelgen.llm.prompt;
+
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import freemarker.template.Configuration;
+import freemarker.template.Template;
+
+public class TemplatedPromptGenerator<TImpl extends TemplatedPromptGenerator<?, TSelector>, TSelector> extends PromptGenerator<TImpl, TSelector> {
+    private final Map<TSelector, String> prompts;
```

Comment:
I don't think we want to shadow `prompts` in the base class, unless it needs to differ for some reason.  You've added the `getPrompts` method to the superclass so can you use this everywhere you're currently using this redeclared collection?